src/server/scripts/Spells/spell_warlock.cpp

```diff
@@ -1490,6 +1494,78 @@ class spell_warl_volatile_agony : public SpellScript
     }
 };
 
+// Bilescourge Bombers - 267211
+class spell_warl_bilescourge_bombers : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_WARLOCK_BILESCOURGE_BOMBERS_AREATRIGGER });
+    }
+
+    void HandleLaunch(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetCaster()->GetPosition(), SPELL_WARLOCK_BILESCOURGE_BOMBERS_AREATRIGGER, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringSpell = GetSpell()
+            });
+    }
+
+    void Register() override
+    {
+        OnEffectLaunch += SpellEffectFn(spell_warl_bilescourge_bombers::HandleLaunch, EFFECT_0, SPELL_EFFECT_CREATE_AREATRIGGER);
+    }
+};
+
+// Bilescourge Bombers - 15141
+struct areatrigger_warl_bilescourge_bombers : AreaTriggerAI
+{
+    areatrigger_warl_bilescourge_bombers(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger), _tickTimer(TICK_PERIOD) {}
+
+    static constexpr Milliseconds TICK_PERIOD = Milliseconds(500);
+
+    void OnUpdate(uint32 diff) override
+    {
+        _tickTimer -= Milliseconds(diff);
+
+        while (_tickTimer <= 0s)
+        {
+            if (Unit* caster = at->GetCaster())
+            {
+                if (AreaTrigger* targetAreaTrigger = caster->GetAreaTrigger(SPELL_WARLOCK_BILESCOURGE_BOMBERS))
```

Comment:
it looks like this is the areatrigger of this script. You can use `at` directly, like you did in the if above instead. `at` is always valid in `AreaTriggerAI`, so no need to nullcheck it explicitly.

Reply:
There are two area triggers involved:
- A target area trigger created by Spell 267211.
- An area trigger at the caster's position created by Spell 282248.

The visual effect of the Bilescourge crashing into the ground should start from the caster's position area trigger and move toward the target area trigger, ending with the impact at the target location.

Reply:
aye, thanks for the heads up